azure-kusto-data/src/client.rs

```diff
@@ -104,137 +116,170 @@ impl KustoClient {
         &self.management_url
     }
 
-    pub fn execute_with_options<DB, Q>(
+    pub(crate) fn pipeline(&self) -> &Pipeline {
+        &self.pipeline
+    }
+
+    /// Execute a query against the Kusto cluster.
+    /// The `kind` parameter determines whether the request is a query (retrieves data from the tables) or a management query (commands to monitor and manage the cluster).
+    /// This method should only be used if the query kind is not known at compile time, otherwise use [execute](#method.execute) or [execute_command](#method.execute_command).
+    /// # Example
+    /// ```no_run
+    /// use azure_kusto_data::prelude::*;
+    /// # #[tokio::main] async fn main() -> Result<(), Error> {
+    ///
+    /// let client = KustoClient::new(
+    ///   ConnectionString::with_default_auth("https://mycluster.region.kusto.windows.net/".to_string()),
+    ///   KustoClientOptions::default())?;
+    ///
+    ///  // Once the [IntoFuture] trait is stabilized, we can drop the call the `into_future()` here
+    ///  let result = client.execute_with_options("some_database".into(), ".show version".into(), QueryKind::Management, None).into_future().await?;
+    ///
+    /// assert!(matches!(result, KustoResponse::V1(..)));
+    /// # Ok(())}
+    /// ```
+    #[must_use]
+    pub fn execute_with_options(
         &self,
-        database: DB,
-        query: Q,
+        database: String,
+        query: String,
```

Comment:
Personally I like trait based parameters - esecially when i started with rust there were a lot of conversions into strings etc going on ... `Into<String>` however is indeed maybe not the most convenient choice, but maybe we could do `AsRef<str>`. Lots of types implement that, and it's quite convenient from the usage perspective.

Reply:
I thought it makes it a bit too verbose, but then I remembered the impl Trait syntax and now it's very lean.
It does mean that some methods can't be const, but it's probably good enough for now.
WDYT?

Reply:
agreed!